Do not use -Werror in the default CFLAGS

Newer compilers can suddenly emit warnings about things that are not fixed in the upstream repo yet, but the build will fail.

Please remove the flag as most distribution will patch it out anyway in order to perform a working build.

(my current warning:

```
driver/kvm/kvm.c: In function ‘kvm_put_memory’:
driver/kvm/kvm.c:1216:9: error: ignoring return value of ‘write’, declared with attribute warn_unused_result [-Werror=unused-result]
         write(kvm_get_instance(vmi)->socket_fd, buf, length);
         ^
driver/kvm/kvm.c:1217:9: error: ignoring return value of ‘read’, declared with attribute warn_unused_result [-Werror=unused-result]
         read(kvm_get_instance(vmi)->socket_fd, &status, 1);
         ^
driver/kvm/kvm.c: In function ‘destroy_domain_socket’:
driver/kvm/kvm.c:318:9: error: ignoring return value of ‘write’, declared with attribute warn_unused_result [-Werror=unused-result]
         write(kvm->socket_fd, &req, sizeof(struct request));
         ^
cc1: all warnings being treated as errors
```

)

--- a/configure.ac
+++ b/configure.ac
@@ -141,6 +141,13 @@ AC_ARG_ENABLE([examples],
       [enable_examples=$enableval],
       [enable_examples=yes])
 
+AC_ARG_ENABLE([debug],
+  [AS_HELP_STRING([--enable-debug],
+    [Enable strict compiler checks @<:@no@:>@])],
+  [debug="$enableval"],
+  [debug="no"])
+AM_CONDITIONAL([DEBUG], [test x$debug = xyes])
+
 dnl -----------------------------------------------
 dnl Checks for programs, libraries, etc.
 dnl -----------------------------------------------

--- a/examples/Makefile.am
+++ b/examples/Makefile.am
@@ -3,7 +3,13 @@
 SUBDIRS =
 
 AM_CFLAGS = -I$(top_srcdir) $(GLIB_CFLAGS)
-AM_CFLAGS += -Werror -Wall -Wextra -Wno-missing-field-initializers -Wno-unused-parameter
+AM_CFLAGS += -Wall -Wextra
+
+if DEBUG
+AM_CFLAGS += -Werror
+AM_CFLAGS += -Wno-missing-field-initializers -Wno-unused-parameter
+endif
+
 AM_LDFLAGS = -L$(top_builddir)/libvmi/.libs/
 LDADD = -lvmi -lm $(LIBS) $(GLIB_LIBS)
 

--- a/libvmi/Makefile.am
+++ b/libvmi/Makefile.am
@@ -111,6 +111,10 @@ lib_LTLIBRARIES= libvmi.la
 libvmi_la_SOURCES= $(h_public) $(h_private) $(drivers) $(os) $(c_sources)
 libvmi_la_LIBADD= config/libconfig.la
 libvmi_la_CFLAGS= -fvisibility=hidden $(GLIB_CFLAGS)
-libvmi_la_CFLAGS+= -Werror -Wall -Wextra
-libvmi_la_CFLAGS+= -Wno-missing-field-initializers -Wno-unused-but-set-variable
+libvmi_la_CFLAGS+= -Wall -Wextra
 libvmi_la_LDFLAGS= -release $(RELEASE) $(GLIB_LIBS)
+
+if DEBUG
+libvmi_la_CFLAGS+= -Werror
+libvmi_la_CFLAGS+= -Wno-missing-field-initializers -Wno-unused-but-set-variable
+endif

